### PR TITLE
criu: 3.15 -> 3.17.1

### PR DESCRIPTION
**Pull request by fpletz**
###### Description of changes

https://criu.org/Download/criu/3.16
https://criu.org/Download/criu/3.17

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/crun/default.nix [hunk 1]
```diff
@@ -11,6 +11,7 @@
 , yajl
 , nixosTests
 , criu
+, protobufc
```
Comment (zowoq) [inline pkgs/applications/virtualization/crun/default.nix:14]: Please move the crun changes to a separate commit.

Comment (zowoq) [reply pkgs/applications/virtualization/crun/default.nix:14]: Also, why is it being added to crun? If it's needed for criu shouldn't it be propagated from there?

Comment (fpletz) [reply pkgs/applications/virtualization/crun/default.nix:14]: It is needed by crun with the new criu version but lxd doesn't need it even though it depends on criu? Not sure if this really needs to be propagated from criu.

Comment (fpletz) [reply pkgs/applications/virtualization/crun/default.nix:14]: Ah, lxd only uses the binaries and calls them. 

```
In file included from /nix/store/sfg8lcmcgq7fm3p7aafrgdzbl5879rfs-criu-3.17.1-dev/include/criu/criu.h:25,
                 from src/checkpoint.c:30:
/nix/store/sfg8lcmcgq7fm3p7aafrgdzbl5879rfs-criu-3.17.1-dev/include/criu/rpc.pb-c.h:7:10: fatal error: protobuf-c/protobuf-c.h: No such file or directory
    7 | #include <protobuf-c/protobuf-c.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

Yeah, I overlooked that the criu headers include protobufc so this needs to be propagated.

Comment (zowoq) [reply pkgs/applications/virtualization/crun/default.nix:14]: Unless it is a explicit (and documented) dependency of crun I don't think it should the added there.

**Comment by fpletz** [comment]
Also removed prefixing PATH since iptables can be passed with the CR_IPTABLES environment variable: https://github.com/checkpoint-restore/criu/blob/7d7d25f946e10b00c522dc44eb9c60d9eba2e7a0/criu/net.c#L1974
